### PR TITLE
Update TestQoSContainerCgroup test

### PR DESCRIPTION
**Pull request by wqld**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR:

* Update TestQoSContainerCgroup test for coverage
  * 17.4 -> 46.4

#### Which issue(s) this PR fixes:

It is a part of #109717 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/cm/qos_container_manager_linux_test.go [hunk 1]
```diff
@@ -310,7 +310,6 @@ func TestQosContainerManagerImpl_Start(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			mockCtrl := gomock.NewController(t)
-			defer mockCtrl.Finish()
```
Comment (21kyu) [inline pkg/kubelet/cm/qos_container_manager_linux_test.go]: Delete this line according to the information provided by @endocrimes at https://github.com/kubernetes/kubernetes/pull/109746#discussion_r864559039.
Thank you!

**Comment by 21kyu** [comment]
/hold
Actually I don't want this PR to be merged like this at all.
If possible, I'd like to know the implicit rules for the scope of objects to be mocked in unit tests that will run inside k8s repo.

**Comment by 21kyu** [comment]
I created this PR to get some advice (I thought I'd get feedback quickly), but I'll try to fix it in a way that doesn't use mocked objects.

**Comment by 21kyu** [comment]
/remove sig/api-machinery sig/architecture sig/auth sig/cli sig/cloud-provider sig/cluster-lifecycle sig/instrumentation sig/network sig/scheduling sig/storage sig/testing area/apiserver area/cloudprovider area/code-generation area/dependency area/e2e-test-framework area/kubeadm area/kubectl area/test

**Comment by 21kyu** [comment]
/remove-sig api-machinery architecture auth cli cloud-provider cluster-lifecycle instrumentation network scheduling storage testing

**Comment by 21kyu** [comment]
/remove-area apiserver cloud provider code-generation dependency e2e-test-framework kubeadm kubectl test

**Comment by 21kyu** [comment]
I'm sorry.
This PR is too messy, so I'll close it and make a new PR.

**Comment by 21kyu** [comment]
I'm sorry.
I made a mistake while trying to synchronize with the upstream, so this PR is very messy.. 😢

**Comment by 21kyu** [comment]
/assign @dchen1107
Please take a look, thank you.

**Comment by pacoxu** [comment]
The adding test LGTM.

However, QOSContainerManager is the interface to be test. The key functions are not mentioned in the UT, this seems to be a little odd to me.

```
type QOSContainerManager interface {
	Start(func() v1.ResourceList, ActivePodsFunc) error
	GetQOSContainersInfo() QOSContainersInfo
	UpdateCgroups() error
}
```

**Comment by 21kyu** [comment]
@pacoxu Thank you for your review!

It's been a while, but to recall,
I wanted to write a unit test for all the methods of QOSContainerManger.
However, the methods of that interface internally created and updated group through runc, and it seems that it needs sudo permission to access the root path of cgroup. (e.g., /sys/fs/cgroup)

I thought that internal parts such as runc could be handled by GoMock in terms process of creating a unit test for QOSContainerManager, so I tried to get advice from this PR, but I couldn't get any reply.

This decision-making is still a daunting task for me, I put this on the back burner and started writing tests to increase coverage.

Please let me know if I'm mistaken or have a good idea. I still want to reinforce this unit test if I have time and a way.

**Comment by 21kyu** [comment]
May I close this PR?

**Pull request closed without merging**